---
title: "Annotating transcriptome from Oxford Nanopore Technology data"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Intro intro.

# Import custom transcriptome into R
Text text.

```{r importdata}
library(rtracklayer)
ont.gtf <- import("ont_merged.gtf.gz")
```


```{r previewdata}
head(ont.gtf)
```


# Download reference annotation and genome
Text text

```{r loadref}
# load library
library(AnnotationHub) 
ah <- AnnotationHub() 

# search database for latest mouse annotation from gencode
query(ah, c("Mus musculus","gencode", "gff")) 
ref.gtf <- ah[['AH49546']] 

# search for mm10 genome sequence in 2bit format 
query(ah, c("mm10","2bit")) 
Mmusculus <- ah[['AH14005']] 
```


# Match chromosomes and genes from custom transcriptome to reference
Text Text

```{r matchinfo}
library(factR)

ont.gtf <- matchChromosomes(ont.gtf, ref.gtf)
ont.gtf <- matchGeneInfo(ont.gtf, ref.gtf, primary_gene_id = "gene_id", secondary_gene_id = "ref_gene_id")
```

# Preview transcripts
Text Text

```{r previewtx}
viewTranscripts(ont.gtf, gene_name == "Ptbp1")
```



# Filter *de novo* transcripts
Text text

```{r subsetnewtranscripts}
ont_denovo.gtf <- subsetNewTranscripts(ont.gtf, ref.gtf)

# compute number of denovo transcripts in transcriptome
length(unique(ont_denovo.gtf$transcript_id)) #denovo
length(unique(ont.gtf$transcript_id)) #all

```


# Build CDS
Text Text

```{r buildcds}
ont_denovo.gtf <- buildCDS(ont_denovo.gtf, ref.gtf, Mmusculus)

```


# Identify protein domains
# Predict NMD-sensitive transcripts
Text text

```{r}
NMDprediction.out <- predictNMD(ont_denovo.gtf)
```

